diff --git a/dark_mode_switch.libraries.yml b/dark_mode_switch.libraries.yml
index fd93609..70897cd 100644
--- a/dark_mode_switch.libraries.yml
+++ b/dark_mode_switch.libraries.yml
@@ -2,4 +2,6 @@ dark_mode_switch:
   js:
     js/dark_mode_switch.js: {}
   dependencies:
+    - core/jquery
     - core/jquery.cookie
+    - core/jquery.once
diff --git a/js/dark_mode_switch.js b/js/dark_mode_switch.js
index 669f3ca..1e6fbf9 100644
--- a/js/dark_mode_switch.js
+++ b/js/dark_mode_switch.js
@@ -5,14 +5,18 @@
       if (context === document) {
         // Check Dark Mode Switch Cookie.
         $(context).find('#dark-mode-switch').once('dark_mode_switch').each(function () {
-          var darkmodestate = ($.cookie('darkmodeswitch') === undefined) ? false : JSON.parse($.cookie('darkmodeswitch'));
-          $('body').toggleClass('darkmode', darkmodestate);
+          var darkmodestate = window.matchMedia('(prefers-color-scheme: dark)').matches;
+          if ('theme' in localStorage) {
+            darkmodestate = (localStorage.theme === 'false') ? false : true;
+          }
+          localStorage.theme = darkmodestate;
+          $('body').toggleClass('dark', darkmodestate);
           $('#dark-mode-switch').prop("checked", darkmodestate);
         });
         // Toggle Dark Mode Switch.
         $('#dark-mode-switch').change(function () {
-          $('body').toggleClass('darkmode');
-          $.cookie('darkmodeswitch', $(this).is( ":checked" ), { expires: 365, path: '/' });
+          $('body').toggleClass('dark');
+          localStorage.theme = $(this).prop("checked");
         });
       }
     }
